documentation/src/test/java/org/hibernate/search/documentation/mapper/orm/automaticindexing/HibernateOrmAutomaticIndexingFilterIT.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.documentation.mapper.orm.automaticindexing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmUtils.with;
+
+import javax.persistence.Basic;
+import javax.persistence.Entity;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+
+import org.hibernate.search.documentation.testsupport.BackendConfigurations;
+import org.hibernate.search.documentation.testsupport.DocumentationSetupHelper;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.GenericField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HibernateOrmAutomaticIndexingFilterIT {
+
+	@Rule
+	public DocumentationSetupHelper setupHelper = DocumentationSetupHelper.withSingleBackend(
+			BackendConfigurations.simple() );
+
+	private EntityManagerFactory entityManagerFactory;
+
+	@Before
+	public void setup() {
+		this.entityManagerFactory = setupHelper.start()
+				.setup( EntityA.class, EntityExtendsA1.class, EntityExtendsA2.class );
+	}
+
+	@Before
+	public void clearFilter() throws Exception {
+		Search.automaticIndexingFilter(
+				entityManagerFactory,
+				ctx -> { /*clear out any settings from tests*/ }
+		);
```

Comment:
I don't believe this should be necessary since a new entity factory will be created for each test method.

Reply:
Right... it will go away when I move the app filter into search mapping. With what's in the current changes - we'd store a filter in a static variable, and that would not clean up itself (so I suppose one more reason not to do it the way it's done 😃 🙈)